### PR TITLE
Add section on API Discovery

### LINKED ISSUE #2426: Add `np.lookfor` as a discoverability tip to the User Guide
scikit-image's package structure is problematic for functions that span categories. For example, `watershed` segmentation is in morphology, because it is traditionally viewed as a morphological transformation. Conversely, `find_boundaries` is in segmentation, but some users have [looked for it](https://github.com/scikit-image/scikit-image/pull/2396#issuecomment-270022280) in morphology.

I've advocated for creating a flat scikit-image namespace before (https://github.com/scikit-image/scikit-image/issues/1608), but that discussion stalled when `np.lookfor` was [mooted](https://github.com/scikit-image/scikit-image/issues/1608#issuecomment-121798358) as an 80% solution.

However, that is a solution only for those who know about it. So, I propose that we add a section on API discovery to the [user guide](scikit-image.org/docs/dev/user_guide.html).

```python
import numpy as np
np.lookfor('boundaries', 'skimage')
```
or
```python
import skimage
np.lookfor('boundaries', skimage)
```

### PR DESCRIPTION
**Pull request by souravsingh**
## Description
Fixes #2426

### CODE REVIEW

**Comment by souravsingh** [comment]
@jni Is the patch ready for merging?

File: doc/source/user_guide/getting_help.rst [hunk 1]
```diff
@@ -28,6 +28,24 @@ The ``quick search`` field located in the navigation bar of the html
 documentation can be used to search for specific keywords (segmentation,
 rescaling, denoising, etc.).
 
+API Discovery
+-------------
+
+The users can make use of ``np.lookfor`` method to check the functions 
+in scikit-image. You will need NumPy to make use of the ``lookfor`` method.
+Usage Examples-
```
Comment (soupault) [inline doc/source/user_guide/getting_help.rst]: What is this trailing dash for?

File: doc/source/user_guide/getting_help.rst [hunk 2]
```diff
@@ -28,6 +28,24 @@ The ``quick search`` field located in the navigation bar of the html
...
+
+``
```
Comment (soupault) [inline doc/source/user_guide/getting_help.rst]: Use triple ` (see https://github.com/souravsingh/scikit-image/blob/a2757d0a66a80451f13938d550dbc3b28ac47659/doc/source/user_guide/getting_help.rst).

File: doc/source/user_guide/getting_help.rst [hunk 3]
```diff
@@ -28,6 +28,24 @@ The ``quick search`` field located in the navigation bar of the html
...
+import numpy as np
+np.lookfor('boundaries', 'skimage')
+``
+
+Alternatively-
```
Comment (soupault) [inline doc/source/user_guide/getting_help.rst]: If you give an alternative, please, describe the difference between two approaches.

File: doc/source/user_guide/getting_help.rst [hunk 4]
```diff
@@ -28,6 +28,24 @@ The ``quick search`` field located in the navigation bar of the html
 documentation can be used to search for specific keywords (segmentation,
 rescaling, denoising, etc.).
 
+API Discovery
+-------------
+
+The users can make use of ``np.lookfor`` method to check the functions
```
Comment (soupault) [inline doc/source/user_guide/getting_help.rst]: Hmm... I don't think that it is a good idea to call it `method`. `function` should be better.

Comment (soupault) [inline doc/source/user_guide/getting_help.rst]: "to check the functions in scikit-image" <- Maybe rephrase and/or enrich this?
I've never used this function, and the result is not clear to me at all:
```
In [8]: np.lookfor('boundaries', skimage)
Search results for 'boundaries'
-------------------------------
skimage.segmentation.find_boundaries
    Return bool array where boundaries between labeled regions are True
skimage.segmentation.mark_boundaries
    Return image with boundaries between labeled regions highlighted.
skimage.future.graph.rag_boundary
    Comouter RAG based on region boundaries
skimage.segmentation.boundaries._find_boundaries_subpixel
    Notes
skimage.transform.warp
    Warp an image according to a given coordinate transformation.
skimage.transform.swirl
    Perform a swirl transformation.
skimage.filters.gaussian
    Multi-dimensional Gaussian filter
skimage.transform.resize
    Resize image to match a certain size.
skimage.transform.rotate
    Rotate image by a certain angle around its center.
skimage.transform.rescale
    Scale image by a certain factor.
skimage.morphology.watershed
    Return a matrix labeled using the watershed segmentation algorithm
```

Comment (jni) [inline doc/source/user_guide/getting_help.rst]: Yes this paragraph can be improved. A loose suggestion:

```
NumPy provides a ``lookfor`` function to search APIs. By default ``lookfor`` will search the NumPy API:

> NumPy lookfor example, e.g. np.lookfor('eigenvector')

But it can also be used to search in modules, by passing either the module name as a string:

> np.lookfor('boundaries', 'skimage')

or the module itself:

> import skimage
> np.lookfor('boundaries', skimage)
```

Comment (jni) [inline doc/source/user_guide/getting_help.rst]: There might also be a way to use intersphinx to link directly to the numpy lookfor documentation...

**Comment by jni** [comment]
@souravsingh thanks for the PR! This will be a valuable addition, but let's take a bit of extra time to get it right!

**Comment by souravsingh** [comment]
I will close the PR since I mistakenly deleted my branch in which I made the changes. Will submit another PR with the requested changes.

**Comment by jni** [comment]
@souravsingh are you going to resubmit this?
